### PR TITLE
EnchantStyleCustomizer rewrite

### PR DESCRIPTION
**Pull request by romangraef**
This commit does not yet include support for chroma.

### CODE REVIEW

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/item/enchants/EnchantMatcher.java [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+		try {
+			patternWithLevels = Pattern.compile(
+				"(§b|§9|§([b9l])§d§l)(?<" + GROUP_ENCHANT_NAME + ">" + enchantName + ") " +
+					"(?<" + GROUP_LEVEL + ">\\d+|" + romanNumerals + ")(?:(?:§9)?,|(?: §8(?:,?[0-9]+)*)?$)");
+		} catch (PatternSyntaxException e) {
+			NotEnoughUpdates.LOGGER.warn("Invalid pattern constructed for enchant matching", e);
+			return Optional.empty();
+		}
+
+		String formatting =  "§" + colourCode;
```
Comment (hannibal002) [inline src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/item/enchants/EnchantMatcher.java:95]: how many brackets do we need?

Comment (romangraef) [reply src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/item/enchants/EnchantMatcher.java:95]: wdym, there are no brackets in that line

**Comment by hannibal002** [comment]
Unused imports in ItemTooltipListener

File: src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/item/enchants/EnchantMatcher.java (hunk 1 above)
Comment (hannibal002) [reply src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/item/enchants/EnchantMatcher.java:95]: wdym brackets? I'm talking blanks :)

**Comment by NopoTheGamer** [comment]
image
image
using this preset: https://canary.discord.com/channels/516977525906341928/1008003785035096156/1008017633066635334

**Comment by romangraef** [comment]
@NopoTheGamer did you have the same preset in the first screenshot? I know what causes the character duplication

**Comment by NopoTheGamer** [comment]
yes

**Review by hannibal002 — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Review by hannibal002 — DISMISSED** [review]

**Comment by NopoTheGamer** [comment]
image
image
image
a few sets of enchants just dont seem to be coloured? (the bottom rule is just all the ult enchants)
using the same preset as last time

**Comment by romangraef** [comment]
@NopoTheGamer do you want me to change anything or investigate, because if yes id like to have a reproducible preset for this (and if not i would like you to approve)

**Comment by romangraef** [comment]
While this isnt perfectly in sync with the old parser, i feel like these precendence rules are more appropriate

**Review by hannibal002 — DISMISSED** [review]
FFS

**Review by NopoTheGamer — DISMISSED** [review]

**Review by Lulonaut — DISMISSED** [review]

**Review by hannibal002 — APPROVED** [review]
Please stop dismissing my approval 😢

**Review by Lulonaut — APPROVED** [review]

**Review by NopoTheGamer — APPROVED** [review]

**Pull request merged**